rsocket-core/src/main/java/io/rsocket/core/RSocketConnector.java

```diff
@@ -72,53 +94,107 @@
 
   private RSocketConnector() {}
 
+  /**
+   * Static factory method to create an {@code RSocketConnector} instance and customize default
+   * settings before connecting. To connect only, use {@link #connectWith(ClientTransport)}.
+   */
   public static RSocketConnector create() {
     return new RSocketConnector();
   }
 
+  /**
+   * Static factory method to connect with default settings, effectively a shortcut for:
+   *
+   * <pre class="code">
+   * RSocketConnector.create().connectWith(transport);
+   * </pre>
+   *
+   * @param transport the transport of choice to connect with
+   * @return a {@code Mono} with the connected RSocket
+   */
   public static Mono<RSocket> connectWith(ClientTransport transport) {
     return RSocketConnector.create().connect(() -> transport);
   }
 
+  /**
+   * Provide a {@code Payload} with data and/or metadata for the initial {@code SETUP} frame.
+   *
+   * <p>For metadata encoding, consider using one of the following encoders:
+   *
+   * <ul>
```

Comment:
I guess it worth moving that section to `metadataMimeType` method since `CompositeMetadata` works only in case the specific metadata mime-type was set. 

WDYT on that?

Reply:
Good idea.